src/com/maddyhome/idea/vim/ex/handler/JoinLinesHandler.java

```diff
@@ -46,6 +46,6 @@ public boolean execute(@NotNull Editor editor, @NotNull DataContext context, @No
     TextRange range = cmd.getTextRange(editor, context, true);
     range = new TextRange(range.getStartOffset(), range.getEndOffset() - 1);
 
-    return VimPlugin.getChange().deleteJoinRange(editor, range, spaces);
+    return VimPlugin.getChange().deleteJoinRange(editor, editor.getCaretModel().getPrimaryCaret(), range, spaces);
```

Comment:
Why don't you join lines for every caret here?

Reply:
Per-caret handlers are not written in this PR yet. I will add a TODO there.